src/App.tsx

```diff
@@ -1,14 +1,53 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { filterTodos } from './helpers';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [filter, setFilter] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(result => {
+        setTodosFromServer(result);
+        setIsLoading(false);
+      });
+  }, []);
+
+  const visibleTodos = filter !== 'all' || query
+    ? filterTodos(todosFromServer, filter, query)
+    : todosFromServer;
+
+  const selectedTodo = visibleTodos.find(todo => todo.id === selectedTodoId) || null;
```

Comment:
Maybe we can set full todo in the state instead of only id, to not search it every time?